crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -884,6 +895,41 @@ impl<'a> Resolver<'a> {
                 }
                 Literal::Integer(integer) => HirLiteral::Integer(integer),
                 Literal::Str(str) => HirLiteral::Str(str),
+                Literal::FmtStr(str) => {
+                    let re = Regex::new(r"\{([\S]+)\}").expect(
+                        "ICE: an invalid regex pattern was used for checking format strings",
+                    );
+                    if re.is_match(&str) {
+                        let mut fmt_str_idents = Vec::new();
+                        for field in re.find_iter(&str) {
```

Comment:
Do you need to explicitly check for a match? Doesn't the not-match case also work for the "iterated 0 times in re.find_iter` case?

Reply:
you are right I don't need to do so. this must be leftover from an earlier iteration of the PR. I don't use `is_match` anymore